projects/dsp-ui/README.md

```diff
@@ -1,26 +1,288 @@
+import { environment } from './environments/environment';
+
+if (environment.production) {
+  enableProdMode();
+}
+
+function bootstrapFailed(result) {
+  console.error('bootstrap-fail! Config is missing or api parameters are not defined', result);
+}
+
+fetch(`config/config.${environment.name}.json`)
+  .then(response => response.json())
+  .then(config => {
+    if (!config) {
+      bootstrapFailed(config);
+      return;
+    }
+
+    // store the response somewhere that the AppInitService can read it.
+    window['tempConfigStorage'] = config;
+
+    platformBrowserDynamic()
+      .bootstrapModule(AppModule)
+      .catch(err => bootstrapFailed(err));
+  })
+  .catch(bootstrapFailed);
+
+```
+
+## Usage
+<!-- TODO: add the modules to app.modules and use them as usual  -->
+<!-- app.modules -->
+Add the desired modules from DSP-UI to the `app.module.ts`. At least `DspCoreModule` has to be imported!
+
+```typescript
+@NgModule({
+  declarations: [
+    AppComponent,
+    ProjectsComponent
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    DspCoreModule        // <-- add the dsp-ui core module here
+  ],
+  providers: [ ... ]    // <-- add providers as mentioned in section above
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+<!-- example of component e.g. get all projects and display as a list -->
+The **DspCoreModule** is a configuration handler for [`@knora/api`](https://www.npmjs.com/package/@knora/api) which offers all the services to make [DSP-API requests](https://docs.dasch.swiss/developers/knora/api-reference/queries/). The following projects-component example shows how to implement the two libraries to get all projects form DSP-API:
+
+```typescript
+import { Component, Inject, OnInit } from '@angular/core';
+import { DspApiConnectionToken } from '@dasch-swiss/dsp-ui';
+import { ApiResponseData, ApiResponseError, KnoraApiConnection, ProjectsResponse, ReadProject } from '@knora/api';
+
+@Component({
+  selector: 'app-projects',
+  template: `<ul><li *ngFor="let p of projects">{{p.longname}} (<strong>{{p.shortname}}</strong> | {{p.shortcode}})</li></ul>`
+})
+export class ProjectsComponent implements OnInit {
+  projects: ReadProject[];
+
+  constructor(
+    @Inject(DspApiConnectionToken) private dspApiConnection: KnoraApiConnection
+  ) { }
+
+  ngOnInit() {
+    this.getProjects();
+  }
+
+  getProjects() {
+    this.dspApiConnection.admin.projectsEndpoint.getProjects().subscribe(
+      (response: ApiResponseData<ProjectsResponse>) => {
+        this.projects = response.body.projects;
+      },
+      (error: ApiResponseError) => {
+        console.error(error);
+      }
+    );
+  }
+}
+```
+
+The **DspViewerModule** contains components to display resources; as single one or as a list for search results. It comprises resource sub-components such as file representations components to display still image, video, audio or text only. But also value components to use single property elements.
```

Comment:
lists are not implemented yet

Reply:
could be listed as a future step (roadmap)

Reply:
list of resources in search results is not implemented yet? okay... This will be part of moving from old knora-ui. Correct?

Reply:
> ist of resources in search results is not implemented yet? okay... This will be part of moving from old knora-ui. Correct?

I think it is enough to mention that this feature will be added in a future release (roadmap)

Reply:
I didn't know that it's not implemented yet. Thanks for the hint

Reply:
BTW: it's a release candidate and this feature will be implemented in the upcoming release candidate .2 or .3?

Reply:
It should be in the roadmap. So the roadmap should probably we public too so we can link to it

Reply:
or we can make a simpler version of the roadmap for the public

Reply:
It has to go to the documentation docs.dasch.swiss or at least as a link to the public (and simplified) google docs file. I agree